### PR TITLE
[WFLY-18492] messaging-clustering-singleton Quickstart Common Enhance…

### PR DESCRIPTION
**Pull request by istudens**
…ments CY2023Q3

Issue: https://issues.redhat.com/browse/WFLY-18492

### CODE REVIEW

File: messaging-clustering-singleton/charts/helm.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+build:
```
Comment (emmartins) [inline messaging-clustering-singleton/charts/helm.yaml:1]: Not needed? Considering this is not OS compatible quickstart...

File: messaging-clustering-singleton/README-source.adoc [hunk 2]
```diff
@@ -468,20 +500,78 @@ This script removes the server configuration that was done by the install-standa
...
+
+Instead of using a standard {productName} server distribution, the two {productName} servers to deploy and run the quickstart can be alternatively provisioned by activating the Maven profile named `provisioned-server` when building the quickstart:
+[source,sh,subs="+quotes,attributes+",options="nowrap"]
+----
+$ mvn clean package -Pprovisioned-server
+$ mvn package -Pprovisioned-server \
+  -Dwildfly.provisioning.dir=server2 -Djboss-as.home=target/server2
+----
+
+The provisioned {productName} servers, with the quickstart deployed, can then be found in the `target/server` directory, and their usage is similar to a standard server distribution, with the simplification that there is never the need to specify the server configuration to be started.
```
Comment (emmartins) [inline messaging-clustering-singleton/README-source.adoc:522]: in the `target` directory

File: messaging-clustering-singleton/README-source.adoc [hunk 3]
```diff
@@ -468,20 +500,78 @@ This script removes the server configuration that was done by the install-standa
...
-//*************************************************
-ifdef::ProductRelease[]
+[[build_and_run_the_quickstart_with_provisioned_server]]
+== Building and running the quickstart application with provisioned {productName} server
+
+Instead of using a standard {productName} server distribution, the two {productName} servers to deploy and run the quickstart can be alternatively provisioned by activating the Maven profile named `provisioned-server` when building the quickstart:
+[source,sh,subs="+quotes,attributes+",options="nowrap"]
+----
+$ mvn clean package -Pprovisioned-server
+$ mvn package -Pprovisioned-server \
```
Comment (emmartins) [inline messaging-clustering-singleton/README-source.adoc:518]: I think you could instead configure 2 plugin executions in the provisioned-server profile, so you would provision both using single command 'mvn clean package -Pprovisioned-server'.

For the first execution you would configure <provisioningDir>server1</provisioningDir>, and <provisioningDir>server2</provisioningDir> for the second.

PS: please note "server" to "server1" move that would affect all instructions after too

Comment (istudens) [reply messaging-clustering-singleton/README-source.adoc]: Done. Just a note that I would keep it as `server` and `server2` to be similar with other QSs. WDYT?

File: messaging-clustering-singleton/charts/helm.yaml (hunk 1 above)
Comment (istudens) [reply messaging-clustering-singleton/charts/helm.yaml]: done

File: messaging-clustering-singleton/README-source.adoc (hunk 2 above)
Comment (istudens) [reply messaging-clustering-singleton/README-source.adoc]: done

File: messaging-clustering-singleton/README-source.adoc (hunk 3 above)
Comment (emmartins) [reply messaging-clustering-singleton/README-source.adoc]: fine with me

File: messaging-clustering-singleton/install-standalone.cli [hunk 4]
```diff
@@ -8,11 +8,6 @@ batch
 # -- Add mdb-delivery-group named "my-mdb-delivery-group"
 /subsystem=ejb3/mdb-delivery-group=my-mdb-delivery-group:add(active=true)
 
-# -- Deploy the messaging-clustering-singleton.war file
-# NOTE: The following command assumes the quickstart is built
-#       If that is not the case, run mvn clean install before this script
-deploy target/messaging-clustering-singleton.war
-
 # -- Run this batch file
 run-batch
```
Comment (emmartins) [inline messaging-clustering-singleton/install-standalone.cli:13]: please remove the reload instruction, there is an infamous issue with it and server provisioning, specially on GH actions

File: messaging-clustering-singleton/pom.xml [hunk 5]
```diff
@@ -143,7 +152,106 @@
...
+                                <packaging-script>
+                                    <scripts>
+                                        <script>${basedir}/install-standalone.cli</script>
+                                    </scripts>
+                                    <!-- Expressions resolved during server execution -->
+                                    <resolve-expressions>false</resolve-expressions>
+                                </packaging-script>
+                            </packaging-scripts>
+                            <name>ROOT.war</name>
+                            <layersConfigurationFileName>standalone-full-ha.xml</layersConfigurationFileName>
```
Comment (emmartins) [inline messaging-clustering-singleton/pom.xml:206]: I think (could be wrong) you should not use this, so the config name generated is standalone.xml, which then no longer requires you to select custom config name when using wildfly:start in the README-source.adoc instructions

**Review by emmartins — CHANGES_REQUESTED** [review]
sorry, one more iteration needed

File: messaging-clustering-singleton/pom.xml (hunk 5 above)
Comment (istudens) [reply messaging-clustering-singleton/pom.xml]: done

File: messaging-clustering-singleton/install-standalone.cli (hunk 4 above)
Comment (istudens) [reply messaging-clustering-singleton/install-standalone.cli]: done

**Pull request merged**